[#4] Initial config refactor for write command

vaugenwake opened the pull request:
Basic working version of an init config command to simplify onboarding and connection to your jira account.

**Todo:**

- [ ] Add tests (I am not really sure how we are going to test this TBH)
- [x] Add form validation for inputs like email & URL
- [ ] Add a ping request to validate the config works and we can connect to the API

vaugenwake committed:
[#4] Initial config refactor for write command

- Added a basic form for config options
- Refactor config struct to provide a write method
- Refactor Load() config to use .config instead of project location

vaugenwake commented:
Addresses #4

vaugenwake commented:
@matthewrobinsondev @theoboldalex Thoughts on this? Not to sure about how I have handled the file paths etc.

Basically Viper doe not let you write the config if the file does not already exist, so needed to check for the directory and the file before I tried to write the config definition from Viper to the file.

Also @matthewrobinsondev whats your thoughts on how to approach testing this bad boy? Write a file in the test and assert it and then clean up the file after the test is closing off maybe?

theoboldalex commented on pkg/config/config.go:
```diff
@@ -3,25 +3,77 @@ package config
+}
+
+func (c ConfigService) createIfNotExist() error {
+	if err := os.MkdirAll(c.getBasePath(), 0755); err != nil {
```
const for the file permissions?

matthewrobinsondev commented:
> Also @matthewrobinsondev whats your thoughts on how to approach testing this bad boy? Write a file in the test and assert it and then clean up the file after the test is closing off maybe?

Yah, create a tmp directory through os int he test, write & think` t.cleanup` allows you to do _the cleanup_

vaugenwake committed:
[#4] Fix base client and update readme
- Update README with init instructions
- Fix base client to append API version as needed
- Add config dir permission const

vaugenwake committed:
Merge branch 'master' of github.com:lazyjira/lazyjira into feat/adding-config-command

vaugenwake commented:
Finished the basic working auth/setup command. Also refactored a few bits to be a bit more usable in the future.

Did look at adding tests for the auth bits but for the life of me can't find a decent way to deal with viper

vaugenwake committed:
[#4] Add override confirmation

matthewrobinsondev commented:
going to create an issue for enhancement in the future & not for this PR but I think being able to show the config would be easy & useful

matthewrobinsondev commented on readme.md:
```diff
@@ -1,4 +1,25 @@
 # Work In Progress!
 On the quest to never leave the terminal.
 
-**Code before POs**
```
smh

matthewrobinsondev commented:
gonna merge
